examples/manifold/plot_t_sne_perplexity.py

```diff
@@ -0,0 +1,104 @@
+"""
+=============================================================================
+ t-SNE: The effect of various perplexity values on the shape
+=============================================================================
+
+An illustration of t-SNE on the two concentric circles and the S-curve
+datasets for different perplexity values.
+
+We observe the tendency of higher variations for small perplexity values and
+clearer shapes as the value increases.
+
+The size, the distance and the shape of clusters may vary upon initialization,
+perplexity values and does not always convey a meaning.
+
+As shown below, t-SNE for higher perplexities finds meaningful topology of
+two concentric circles, however the size and the distance of the circles varies
+slightly from the original. In contrary to two circles dataset, the shapes
+visually diverge from S-curve topology on S-curve dateset even for larger
+perplexity values.
+
+For further details, "How to Use t-SNE Effectively"
+http://distill.pub/2016/misread-tsne/ provides a good discussion of the
+effects of various parameters, as well as interactive plots to explore
+those effects.
+"""
+
+# License: BSD 3 clause
+
+print(__doc__)
+
+import matplotlib.pyplot as plt
+
+from matplotlib.ticker import NullFormatter
+from mpl_toolkits.mplot3d import Axes3D
+from sklearn import manifold, datasets
+from time import time
+
+# Next line to silence pyflakes. This import is needed.
+Axes3D
+
+n_samples = 500
+n_components = 2
+subplot_index = 251
```

Comment:
I think it would be clearer to use `add_subplot(2, 5, subplot_index)`, or to use `fig.subplots(2, 5)` to get an array of axes to draw on .